src/Redis/Orleans.Clustering.Redis/Storage/RedisMembershipTable.cs

```diff
@@ -40,11 +40,12 @@ public async Task DeleteMembershipTableEntries(string clusterId)
         public async Task InitializeMembershipTable(bool tryInitTableVersion)
         {
             _muxer = await _redisOptions.CreateMultiplexer(_redisOptions);
-            _db = _muxer.GetDatabase(_redisOptions.Database);
+            _db = _muxer.GetDatabase();
 
             if (tryInitTableVersion)
             {
                 await _db.HashSetAsync(_clusterKey, TableVersionKey, SerializeVersion(DefaultTableVersion), When.NotExists);
+                await _db.KeyExpireAsync(_clusterKey, _redisOptions.EntryExpiry);
```

Comment:
do we actually want to issue this if `EntryExpiry` is `null`? this will result in constantly issuing `PERSIST` instructions, which is redundant if we've never set an expiration

note: expirations in redis are per key, not per hash item - this will tie all hash entries to this expiration - just want to check that's OK

Reply:
Yes, this is only intended for test & maybe dev scenarios. Deleting the entire key is intended.

I suppose we can skip if it's null. If a user accidentally set an expiry, they can manually fix it before it's too late.